### PR TITLE
Add allocation filters

### PR DESCRIPTION
**Pull request by joonatank**
- Add: allocation page filters using query params
- Fix: only query valid application statuses for applicationRound page
- Fix: LinkPrev should always travel up path, not history back
- Fix: ApplicationRounds page showing application in a wrong section (based on status)
- Move: ui implementation of search tags to common (styled and reusable)
- Remove unused application page

Refs: TILA-2923

### CODE REVIEW

File: apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx [hunk 1]
```diff
@@ -45,214 +55,632 @@ const Tab = styled(Tabs.Tab)`
...
+  const setUnitFilter = (value: number) => {
+    // NOTE different logic because values are not atomic and we need to set two params
+    const vals = new URLSearchParams(searchParams);
+    vals.set("unit", value.toString());
+    // TODO how to find the default reservation unit for the selected unit?
+    vals.delete("reservation-unit");
+    setParams(vals);
+  };
+
+  // TODO if unit selection changes and new list doesn't include the selected reservation unit, clear the selection
```
Comment (joonatank) [inline apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx:163]: Remove comment

File: apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx [hunk 2]
```diff
@@ -45,214 +55,632 @@ const Tab = styled(Tabs.Tab)`
...
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- this is the correct list, but should be refactored
+  }, [units]);
+
+  const unitReservationUnits = reservationUnits.filter(
+    (ru) => ru.unit?.pk != null && ru?.unit?.pk === Number(unitFilter)
+  );
+
+  // TODO default value here instead of the JSX
+  // because otherwise we the query returns all the reservation units if the filter is not set
```
Comment (joonatank) [inline apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx:177]: Remove comment

File: apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx [hunk 3]
```diff
@@ -45,214 +55,632 @@ const Tab = styled(Tabs.Tab)`
...
+          : {}),
+        includePreferredOrder10OrHigher:
+          orderFilter != null &&
+          orderFilter.filter((x) => Number(x) > 10).length > 0,
+        ...(nameFilter != null && Number.isNaN(Number(nameFilter))
+          ? { name_Istartswith: nameFilter }
+          : {}),
+        ...(nameFilter != null && !Number.isNaN(Number(nameFilter))
+          ? { textSearch: nameFilter }
+          : {}),
```
Comment (joonatank) [inline apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx:287]: Text search always. Backend problem that is fixed.

File: apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx [hunk 4]
```diff
@@ -45,214 +55,632 @@ const Tab = styled(Tabs.Tab)`
...
-        )
-      )
-      .filter((ru): ru is ReservationUnitByPkType => ru != null) ?? [];
-
-  const reservationUnits: ReservationUnitByPkType[] = uniqBy(allResUnits, "pk")
-    .filter((ru) => unitFilter?.value === ru?.unit?.pk)
-    .filter((ru): ru is ReservationUnitByPkType => ru != null);
+  const orderOptions = Array.from(Array(10).keys())
+    .map((n) => ({
+      value: n + 1,
```
Comment (joonatank) [inline apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx:359]: Test is it 1 or 0

File: apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx [hunk 5]
```diff
@@ -45,214 +55,632 @@ const Tab = styled(Tabs.Tab)`
...
+  // TODO don't use spinners, skeletons are better
+  // also this blocks the sub component query (the initial with zero filters) which slows down the page load
+  if (loading) {
+    return <Loader />;
+  }
+  // TODO improve this (disabled filters if error, notify the user, but don't block the whole page)
+  if (error) {
+    return <p>Error</p>;
+  }
+
```
Comment (joonatank) [inline apps/admin-ui/src/component/recurring-reservations/allocation/index.tsx:639]: FIX

**Review by vincit-matu — APPROVED** [review]
LGTM 👍

**Pull request merged**